app/models/vm_or_template.rb

```diff
@@ -1884,6 +1884,10 @@ def supports_snapshots?
     false
   end
 
+  def supports_provisioning?
+    true
```

Comment:
@h-kataria I understand that you are using the newer terminology so you should be using the `supports_feature_mixin' like here [app/models/manageiq/providers/vmware/infra_manager.rb#L23](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/vmware/infra_manager.rb#L23)?   cc @durandom

More importantly I think setting this to `true` at the base level is incorrect.  Having it here says that all VMs support provisioning (since VM model is a subclass of VmOrTemplate) which is incorrect.  We only support provisioning of VMs on the VMware provider, all others should return false.  And there are other conditions to consider like the check you are adding for template to ensure it is associated to a provider.  Any new providers added would likely not support provisioning immediately so they would also have know to disable this flag.  It seems more likely it should be disabled by default and enabled per-provider as support is added.

Then you also need to address the MiqTemplate sub-classes.  There two I do not think it makes sense to immediately assume that we support provisioning by setting it at the base level.  In my opinion the base should always say the feature is not supported and each provider would enable it where appropriate.

Does that make sense or is there something I am misunderstanding about intention of this method?